src/Dropdown.php

```diff
@@ -54,11 +54,11 @@ public function onChange(\Closure $fx): void
             ),
         ]);
 
-        $this->cb->set(static function (Jquery $j, string $value) use ($fx) {
+        $this->cb->set(static function (Jquery $j, $value) use ($fx) {
             return $fx($value);
         }, ['item' => new JsCallbackLoadableValue(null, function ($v) {
             return $this->getApp()->uiPersistence->typecastLoadField(
-                $this->model->getField('id'),
+                $this->model->getIdField(),
```

Comment:
`'id'` is hardcoded in https://github.com/atk4/ui/blob/5.2.0/template/dropdown.html#L6 thus `'id'` must be hardcoded here as well - probably related with https://github.com/atk4/ui/pull/2237/files#r1889379601 - I belive this PR cannot work.